test/boost/sstable_datafile_test.cc

```diff
@@ -4175,39 +4174,36 @@ SEASTAR_TEST_CASE(sstable_owner_shards) {
         auto s = builder.build();
 
         auto tmp = tmpdir();
-        auto sst_gen = [&env, s, &tmp, gen = make_lw_shared<unsigned>(1)] () mutable {
-            auto sst = env.make_sstable(s, tmp.path().string(), (*gen)++, la, big);
-            sst->set_unshared();
-            return sst;
-        };
         auto make_insert = [&] (auto p) {
             auto key = partition_key::from_exploded(*s, {to_bytes(p.first)});
             mutation m(s, key);
             m.set_clustered_cell(clustering_key::make_empty(), bytes("value"), data_value(int32_t(1)), 1);
             BOOST_REQUIRE(m.decorated_key().token() == p.second);
             return m;
         };
-
+        auto gen = make_lw_shared<unsigned>(1);
         auto make_shared_sstable = [&] (std::unordered_set<unsigned> shards, unsigned ignore_msb, unsigned smp_count) {
             auto mut = [&] (auto shard) {
                 auto tokens = token_generation_for_shard(1, shard, ignore_msb, smp_count);
                 return make_insert(tokens[0]);
             };
             auto muts = boost::copy_range<std::vector<mutation>>(shards
                 | boost::adaptors::transformed([&] (auto shard) { return mut(shard); }));
-            dht::default_partitioner = std::make_unique<dht::murmur3_partitioner>(1, ignore_msb);
+            auto sst_gen = [&env, s, &tmp, gen, ignore_msb] () mutable {
+                auto schema = schema_builder(s).with_partitioner("org.apache.cassandra.dht.Murmur3Partitioner", 1, ignore_msb).build();
```

Comment:
why is "1" correct here?

Reply:
That's what it was before. See removed "dht::default_partitioner = std::make_unique<dht::murmur3_partitioner>(1, ignore_msb);"

Reply:
Here's what used to be:
We used to:
1. set default_partitioner to dht::murmur3_partitioner(1, ignore_msb)
2. perform all the code below
3. clean up default partitioner to dht::murmur3_partitioner(smp_count, ignore_msb)

We now:
1. Create a new schema with dht::murmur3_partitioner(1, ignore_msb)
2. We perform all the code below with that schema which has the same effect as setting default_partitioner before. We switched all the places that were using the default partitioner to use schema now.
3. We don't have to cleanup default partitioner because we didn't change it.

Reply:
@avikivity are you satisfied by this response? It seems to me he is right that he didn't change the previous behavior, but I don't know if you're trying to understand why it was like this in the first place...

Reply:
Agree that not changing behavior is correct here.